class-gf-razorpay.php

```diff
@@ -587,15 +573,131 @@ public function billing_info_fields()
 
     public function init()
     {
+        wp_register_script('rzp', plugin_dir_url(__FILE__)  . 'assets/js/custom.js', null, null);
+        wp_enqueue_script('rzp');
+
         add_filter( 'gform_notification_events', array( $this, 'notification_events' ), 10, 2 );
 
+        add_action( 'wp_ajax_get_data', 'get_data' );
+        add_action( 'wp_ajax_nopriv_get_data', 'get_data' );
         // Supports frontend feeds.
         $this->_supports_frontend_feeds = true;
 
         parent::init();
 
     }
 
+    public function auto_enable_webhook()
+    {
+            $webhookExist = false;
+            $webhookUrl = esc_url(admin_url('admin-post.php')) . '?action=gf_razorpay_webhook';
+            $enabled = true;
+            $alphanumericString = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ-=~!@#$%^&*()_+,./<>?;:[]{}|abcdefghijklmnopqrstuvwxyz';
+            $secret = substr(str_shuffle($alphanumericString), 0, 20);
+            $getWebhookFlag =  get_option('gf_webhook_enable_flag');
+            $time = time();
+
+            if (empty($getWebhookFlag))
+            {
+                add_option('gf_webhook_enable_flag', $time);
+            }
+            else
+            {
+                update_option('gf_webhook_enable_flag', $time);
+            }
+          
+            $skip = 0;
+            $count = 10;
+            $webhookItems= [];
+
+            do {
+                $webhook = $this->webhookAPI("GET", "webhooks?count=".$count."&skip=".$skip);
+                $skip += 10;
+                if ($webhook['count'] > 0)
+                {
+                    foreach ($webhook['items'] as $key => $value)
+                    {
```

Comment:
@jampulanaveen can we include if ($value['url'] === $webhookUrl) here eg: if there are 30 webhooks and our webhookurl is found in request 1 , request 2 and 3 can b skipped